tests/src/resultcollection.hh

```diff
@@ -0,0 +1,72 @@
+// SPDX-FileCopyrightText: 2022 The Ikarus Developers mueller@ibb.uni-stuttgart.de
+// SPDX-License-Identifier: LGPL-3.0-or-later
+
+#pragma once
+#include "testcommon.hh"
+
+#include <ikarus/finiteelements/ferequirements.hh>
+
+auto createLinearStressResults = []<typename NonLinearOperator, typename FiniteElement>(
+                                     NonLinearOperator& nonLinearOperator, FiniteElement& fe) {
+  constexpr int dim = FiniteElement::Traits::mydim;
+
+  static_assert(std::remove_cvref_t<decltype(fe.localView().tree())>::degree() == dim,
+                "The test to check linear stress is only supported with the powerBasis having the same power as the "
+                "element dimension");
+  static_assert(
+      std::is_same_v<
+          std::remove_cvref_t<decltype(fe.localView().tree().child(0))>,
+          Dune::Functions::LagrangeNode<std::remove_cvref_t<decltype(fe.localView().globalBasis().gridView())>, 1>>,
+      "The test to check linear stress is only supported for a linear Lagrange basis");
+
+  const auto& element          = fe.gridElement();
+  const auto& referenceElement = Dune::referenceElement<double, dim>(element.type());
+  const int nodes              = referenceElement.size(dim);
+
+  Eigen::MatrixXd expectedStress(nodes, dim * (dim + 1) / 2);
+
+  auto& displacement = nonLinearOperator.firstParameter();
+
+  if (element.type() == Dune::GeometryTypes::simplex(FiniteElement::Traits::mydim)) {
+    Eigen::Matrix<double, 1, dim*(dim + 1) / 2> constantStresses;
+    if constexpr (dim == 2) {
+      displacement << 0, 0, 2, 0, 1, 0;
+      constantStresses << 2197.80219780, 659.34065934, 384.61538462;
+      expectedStress << constantStresses, constantStresses, constantStresses;
```

Comment:
Just to let you know about the functionality. You could have written here:
```cpp
expectedStress.rowwise() = constantStresses;
```
see https://eigen.tuxfamily.org/dox-3.2/group__TutorialReductionsVisitorsBroadcasting.html section Broadcasting

but you can decide what is more readable.

Reply:
ah i was looking for this kind of functionality but couldn't find it